src/request.rs

```diff
@@ -165,6 +165,7 @@ pub struct Request<T> {
 /// The HTTP request head consists of a method, uri, version, and a set of
 /// header fields.
 #[derive(Clone)]
+#[non_exhaustive]
```

Comment:
Would you mind making `#[non_exhaustive]` a different PR?

Reply:
Moved this commit to #678.